lib/matplotlib/backend_bases.py

```diff
@@ -2555,6 +2605,123 @@ class NonGuiException(Exception):
     pass
 
 
+class WindowEvent(object):
+    def __init__(self, name, window):
+        self.name = name
+        self.window = window
+
+
+class WindowBase(cbook.EventEmitter):
```

Comment:
Ugh, I dislike ``abc``...
```
File "matplotlib/lib/matplotlib/backends/backend_gtk3.py", line 400, in <module>
    class WindowGTK3(WindowBase, Gtk.Window):
  File "/usr/lib/python2.7/abc.py", line 87, in __new__
    cls = super(ABCMeta, mcls).__new__(mcls, name, bases, namespace)
TypeError: Error when calling the metaclass bases
    metaclass conflict: the metaclass of a derived class must be a (non-strict)
subclass of the metaclasses of all its bases
```
Why do we need to use abc?  I don't see it in use anywhere else in the codebase...